plugins/external_setup_hook/lib/samson_external_setup_hook/samson_plugin.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module SamsonExternalSetupHook
+  class Engine < Rails::Engine
+  end
+
+  POLL_INTERVAL = 15
+  POLL_WAIT_TIME = 30
+  TRIGGER_WAIT_TIME = 30
+
+  def self.wait_for_external_setup(job, output, stage, ctx_data)
+    status_poll_url = nil
+    success = false
+
+    setup_hook = stage.external_setup_hook
+    conn = Faraday.new(:ssl => {:verify => setup_hook.verify_ssl})
+
+    return true if setup_hook.nil?
+
+    output.write("Trigger external setup through #{setup_hook.endpoint}\n")
+
+    trigger_wait_time = TRIGGER_WAIT_TIME
+    until trigger_wait_time < 0 do
+      trigger_wait_time -= POLL_INTERVAL
+
+      response = conn.post(setup_hook.endpoint, ctx_data.to_json) do |req|
+        req.options.timeout = req.options.open_timeout = 5
+        req.headers['Content-Type'] = 'application/json'
+
+        if setup_hook.auth_type == "token"
```

Comment:
maybe just have a single column for "authorization header" where users can enter `Token fooo`
... or otherwise make sure to also support "Bearer" and "Token"